LkeServices/Blockchain/SiriusWalletsService.cs

```diff
@@ -0,0 +1,124 @@
+            TimeSpan retryTimeout,
+            IApiClient siriusApiClient,
+            ILog log)
+        {
+            _brokerAccountId = brokerAccountId;
+            _retryCount = retryCount;
+            _retryTimeout = retryTimeout;
+            _siriusApiClient = siriusApiClient;
+            _log = log;
+
+            _waitAccountCreationPolicy = Policy
+                .HandleResult<AccountSearchResponse>(res =>
+                {
+                    if (res != null && res.ResultCase == AccountSearchResponse.ResultOneofCase.Error)
+                    {
+                        _log.WriteInfo(nameof(CreateWalletsAsync), info: "Error getting account", context: $"error: {res.Error.ToJson()}");
+                        return true;
+                    }
+
+                    var hasAllActiveWallets = res != null && res.Body.Items.Count > 0 && res.Body.Items.All(x => x.State == AccountStateModel.Active);
+                    _log.WriteInfo(nameof(CreateWalletsAsync),info: !hasAllActiveWallets ? "Wallets not ready yet..." : "All wallets are active!", context: null);
+                    return !hasAllActiveWallets;
+                })
+                .WaitAndRetryAsync(_retryCount, retryAttempt => _retryTimeout);
+        }
+
+        public async Task CreateWalletsAsync(string clientId, bool waitForCreation)
+        {
+            var accountResponse = await _siriusApiClient.Accounts.SearchAsync(new AccountSearchRequest
+            {
+                ReferenceId = clientId,
+                Pagination = new PaginationInt64{Limit = 100}
+            });
+
+            if (accountResponse.ResultCase == AccountSearchResponse.ResultOneofCase.Error)
+            {
+                _log.WriteWarning(nameof(CreateWalletsAsync), info: "Error getting wallets from sirius", context: $"Error: {accountResponse.Error.ToJson()}");
+            }
+
+            if (accountResponse.ResultCase == AccountSearchResponse.ResultOneofCase.Body &&
+                accountResponse.Body.Items.Count == 0)
+            {
+                _log.WriteInfo(nameof(CreateWalletsAsync), info: "Creating wallets in sirius", context: clientId);
+
+                var createResponse = await _siriusApiClient.Accounts.CreateAsync(new AccountCreateRequest
+                {
+                    RequestId = $"{_brokerAccountId}{clientId}",
+                    BrokerAccountId = _brokerAccountId,
+                    ReferenceId = clientId
+                });
+
+                if (createResponse.ResultCase == AccountCreateResponse.ResultOneofCase.Error)
+                {
+                    _log.WriteWarning(nameof(CreateWalletsAsync), info: "Error creating wallets in sirius", context: $"Error: {createResponse.Error.ToJson()}");
+                    return;
+                }
+
+                _log.WriteInfo(nameof(CreateWalletsAsync), info: "Wallets created in siruis", context: $"Result: {createResponse.Body.Account.ToJson()}");
+
+                if (waitForCreation)
```

Comment:
We should wait for the creation even if the condition on line 67 is false. To handle the case when the user entered the same page one more time and the account is not activated yet.